src/java/za/co/mip/ablduck/javadoc/DescriptiveErrorListener.java

```diff
@@ -0,0 +1,41 @@
+/**
+ * Copyright 2017 MIP Holdings
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+package za.co.mip.ablduck.javadoc;
+
+import org.antlr.v4.runtime.*;
+
+public class DescriptiveErrorListener extends BaseErrorListener {
+    public static DescriptiveErrorListener INSTANCE = new DescriptiveErrorListener();
+    //TODO: maybe a ant flag to turn this off?
+    private static final boolean REPORT_SYNTAX_ERRORS = true;
+
+    @Override
+    public void syntaxError(Recognizer<?, ?> recognizer, Object offendingSymbol,
+                            int line, int charPositionInLine,
+                            String msg, RecognitionException e)
+    {
+        if (!REPORT_SYNTAX_ERRORS)
+            return;
+
+        String sourceName = recognizer.getInputStream().getSourceName();
+        if (!sourceName.isEmpty()) {
+            sourceName = String.format("%s:%d:%d ", sourceName, line, charPositionInLine);
+        }
+
+        System.err.println(sourceName+" "+msg);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Replace this use of System.out or System.err by a logger. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar.riverside-software.fr/coding_rules#rule_key=squid%3AS106)